src/hssm/distribution_utils/dist.py

```diff
@@ -243,12 +267,32 @@ def rng_fn(
                 **kwargs,
             )
 
-            output = np.column_stack([sim_out["rts"], sim_out["choices"]])
+            sims_out = np.column_stack([sim_out["rts"], sim_out["choices"]])
 
             if not is_all_scalar:
-                output = output.reshape((*max_shape[:-1], max_shape[-1] * n_samples, 2))
+                sims_out = sims_out.reshape(
+                    (*max_shape[:-1], max_shape[-1] * n_samples, 2)
+                )
 
-            return output
+            if p_outlier is not None:
+                assert cls._lapse is not None
+                out_shape = sims_out.shape[:-1]
+                replace = rng.binomial(n=1, p=p_outlier, size=out_shape).astype(bool)
+                replace = np.stack([replace, replace], axis=-1)
+                n_draws = np.prod(out_shape)
+                lapse_rt = pm.draw(
```

Comment:
did you speed test this?
I know usually pm.draw() compile the RV first, just making sure that in this case it doesn't redo that all the time.

Reply:
Ok as per your answer from the previous comment, this should take all draws over all posterior (prior) predictive samples at the same time?
In this case previous point isn't relevant.

Reply:
Just tested with 10 runs in a loop. Seems like each time it takes about the same amount of time. This might be a function of PyMC recompiling the entire computational graph for prior predictive sampling